File: C++/ArrayString/32_MinimumNumPlatform.cpp
#include <bits/stdc++.h>
using namespace std;
int findPlatform(int arr[], int dep[], int n)
{
    int plats = 1;
    int maxPlats = 1;
    sort(arr, arr + n);
    sort(dep, dep + n);
    int i = 1, j = 0;
    while (i < n and j < n)
    {
        if (arr[i] <= dep[j])
        {
            plats++;
            maxPlats = max(maxPlats, plats);
            i++;
        }
        else if (arr[i] > dep[j])
        {
            plats--;
            j++;
        }
    }
    // Your code here
    return maxPlats;
}

File: Java/ArrayString/32_MinimumNumPlatform.java
static int findPlatform(int arr[], int dep[], int n){
        int i = 0, j = 0;
        Arrays.sort(arr);
        Arrays.sort(dep);
        int count = 0, ans = 0;
        while(i<n && j<n){
            if(arr[i] <= dep[j]){
                count++;
                i++;
            }
            else{
                count--;
                j++;
            }
            
            ans = Math.max(ans,count);
        }
        return ans;
    }